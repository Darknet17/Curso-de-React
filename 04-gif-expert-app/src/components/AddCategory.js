import {useState} from "react";
import PropTypes from 'prop-types';
import {Button, Form, FormControl} from "react-bootstrap";

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories(cat=>[inputValue, ...cat])
            setInputValue('');
        }

    }

    return (
        <Form className="d-flex" onSubmit={ handleSubmit }>
            <FormControl
                type="search"
                placeholder="Search"
                className="me-2"
                aria-label="Search"
                value={ inputValue }
                onChange={ handleInputChange }
            />
            <Button type="submit" variant="outline-success">Search</Button>
        </Form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

