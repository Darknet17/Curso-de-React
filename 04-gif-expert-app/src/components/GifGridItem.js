import PropTypes from "prop-types";
import {Card} from "react-bootstrap";

export const GifGridItem = ({ title, url }) => {
    return (
        <Card className="card animate__animated animate__fadeIn" style={{ width: '18rem' }}>
            <Card.Img variant="top" src={ url } />
            <Card.Body>
                <Card.Text> {
                    title
                }</Card.Text>
            </Card.Body>
        </Card>
    );
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url:  PropTypes.string.isRequired
}
