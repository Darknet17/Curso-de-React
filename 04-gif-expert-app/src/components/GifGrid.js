import PropTypes from "prop-types";
import {GifGridItem} from "./GifGridItem";
import {useFetchGifs} from "../hooks/useFetchGifs";
import {Col, Row, Spinner} from "react-bootstrap";

export const GifGrid = ({ category }) => {

    const {data: images, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {  loading && <Spinner animation="grow" variant="dark" />  }

            <Row>
                {
                    images.map(  ({id, title, url }) => (
                        <Col sm={3}>
                            <GifGridItem key={id} title={title} url={url}/>
                        </Col>
                    ))
                }
            </Row>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


