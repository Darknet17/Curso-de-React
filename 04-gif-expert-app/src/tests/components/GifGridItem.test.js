import {shallow} from "enzyme";
import {GifGridItem} from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem />", () => {

    const title = "Un titulo";
    const url = "https://localhost/algo.jpg"
    const wrapper = shallow(<GifGridItem  title={title} url={url}/>) ;

    test("debe de mostrar el componente correctamente", ()=>{
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de tener un parrafo con el titulo", () =>{
        const p = wrapper.find("CardText");
        expect(p.text().trim()).toBe(title);
    });

    test("debe de tener la imagen igual al url y alt de los props", ()=>{
        const img = wrapper.find("CardImg");
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })


    test("debe de tener animate__fadeIn", ()=>{
        const card = wrapper.find("Card");
        const classNames = card.prop("className");
        expect(classNames.includes("animate__fadeIn")).toBe(true);
        //expect(img.prop('class')).toBe();
        //expect(img.prop('alt')).toBe(title);
    })
});
