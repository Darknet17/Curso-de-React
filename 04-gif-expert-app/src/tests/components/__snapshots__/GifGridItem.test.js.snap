// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<Card
  body={false}
  className="card animate__animated animate__fadeIn"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <CardImg
    alt="Un titulo"
    src="https://localhost/algo.jpg"
    variant="top"
  />
  <CardBody>
    <CardText>
       
      Un titulo
    </CardText>
  </CardBody>
</Card>
`;
